src/main/java/com/codeless/api/automation/service/impl/EmailServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.codeless.api.automation.service.impl;
+
+import com.codeless.api.automation.entity.User;
+import com.codeless.api.automation.service.ContentBuilderService;
+import com.codeless.api.automation.service.EmailService;
+import com.codeless.api.automation.service.VerificationService;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import javax.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class EmailServiceImpl implements EmailService {
+
+  private static final String SUBJECT_TEMPLATE = "Codeless API Automation registration verification";
+  private static final String FROM_EMAIL = "api.monitor.bot@gmail.com";
+  private static final String NOTIFICATION_EMAIL_TEMPLATE = "email-verification";
+  private static final String VERIFICATION_URL = "VERIFICATION_URL";
+
+  private final VerificationService verificationService;
+  private final ContentBuilderService contentBuilderService;
+  private final JavaMailSender javaMailSender;
+
+  @Override
+  public void sendEmail(User user, HttpServletRequest httpServletRequest) {
+    MimeMessage mimeMessage = generateVerificationMessage(user, httpServletRequest);
+    log.info("Sending email message to : '{}' ", user.getUsername());
+    javaMailSender.send(mimeMessage);
+  }
+
```

Comment:
To be honest, after considering that in depth, I would say it is better to provide host in property files. What happens if a service is deployed behind a load balancer or API Gateway ?

what is host from fromRequestUri ? is it host where a service is deployed or a load balancer host ?

Let's move it to a property file assuming that we specify a load balancer host or API GW host there.

What is going to be a

Reply:
Email configuration extract to config, Controller and Service methods update to work without http servlet request params